src/main/java/org/traccar/database/OpenIdProvider.java

```
@@ -82,15 +91,47 @@ public OpenIdProvider(Config config, LoginService loginService) {
             baseUrl = new URI(config.getString(Keys.WEB_URL, ""));
+
+            if (
+                config.hasKey(Keys.OPENID_ISSUERURL)
+                && (
+                    !config.hasKey(Keys.OPENID_AUTHURL)
+                    || !config.hasKey(Keys.OPENID_TOKENURL)
+                    || !config.hasKey(Keys.OPENID_USERINFOURL))
+                ) {
+                    HttpClient httpClient = HttpClient.newHttpClient();
+
+                    HttpRequest httpRequest = HttpRequest.newBuilder(
+                        URI.create(
+                            config.getString(Keys.OPENID_ISSUERURL) + "/.well-known/openid-configuration")
+                            )
+                        .header("accept", "application/json")
+                        .build();
+
+                    String httpResponse = httpClient.send(httpRequest, BodyHandlers.ofString()).body();
+
+                    Map<String, Object> discoveryMap = new ObjectMapper().readValue(
+                        httpResponse, new TypeReference<Map<String, Object>>() { });
+
+                    authUrl = new URI(discoveryMap.get("authorization_endpoint").toString());
+                    tokenUrl = new URI(discoveryMap.get("token_endpoint").toString());
+                    userInfoUrl = new URI(discoveryMap.get("userinfo_endpoint").toString());
+
+                    LOGGER.info("OpenID Connect auto discovery successful");
+            } else {
+                authUrl = new URI(config.getString(Keys.OPENID_AUTHURL));
+                tokenUrl = new URI(config.getString(Keys.OPENID_TOKENURL));
+                userInfoUrl = new URI(config.getString(Keys.OPENID_USERINFOURL));
+            }
         } catch (URISyntaxException error) {
             LOGGER.error("Invalid URIs provided in OpenID configuration");
+        } catch (InterruptedException | IOException error) {
+            LOGGER.error("OpenID Connect auto discovery failed");
         }
```

Do we need to catch these? I would just let it crash.